### PR TITLE
Added Ratings feature with option to turn on/off the Ratings feature 

### LINKED ISSUE #639: Ratings in local UI?
First of all, thank you very much for this app! It's been very responsive on my Pi-based server, unlike other ones I've tried.

I'm moving from desktop-hosted content to my server and I'm hoping to get as close to what I'm used to as possible.

I've been using Banshee as a player and I've gotten used to two things: 1) smart playlists (which I see in another ticket that you're planning on implementing [yay!]) and 2) ratings (0-5).

I see in the subsonic code that you support both stars and ratings, but it seems like the local UI only supports stars. Is there a plan to include (or enable by config) ratings?

Thanks again.

### PR DESCRIPTION
**Pull request by neilchauhan2**
## Update with this PR.
- Added Ratings feature to the UI.
- Added option to turn on/off Ratings feature.

## UI Preview with Ratings enabled.
on-s-1
on-s-2
on-al-1
on-al-2
on-al-3
on-al-4
on-a-1

## UI Preview with Ratings disabled.
off-s-1
off-s-2
off-a-1
off-a-2
off-al-1
off-al-2
off-al-3
off-al-4

This PR fixes #639

### CODE REVIEW

**Comment by deluan** [comment]
Thanks for this! I captured a few bugs with this PR:


https://user-images.githubusercontent.com/331353/113656990-084f1680-966b-11eb-9c64-7491fc1a5c36.mov


https://user-images.githubusercontent.com/331353/113656996-08e7ad00-966b-11eb-9390-1abd946be577.mov


Screen Shot 2021-04-05 at 11 53 59 PM

Also, I could not find a way to reset/remove the stars from an item.

**Comment by neilchauhan2** [comment]
@deluan I have resolved the bugs, please review this PR once again. 

> Also, I could not find a way to reset/remove the stars from an item.

This works exactly the same way as in other platforms. You have to click the currently selected star to remove ratings or reset it to 0. For example, a particular song has 3 star ratings, so you will see 3 lighted stars, click on the 3rd lighted star and the ratings would be removed i.e. set back to 0.

File: ui/src/common/RatingField.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
+import React, { useCallback } from 'react'
+import PropTypes from 'prop-types'
+import Rating from '@material-ui/lab/Rating'
+import { makeStyles } from '@material-ui/core/styles'
+import clsx from 'clsx'
+import { useRating } from './useRating'
+
+const useStyles = makeStyles({
+  rating: {
+    visibility: (props) => (props.visible === false ? 'hidden' : 'inherit'),
```
Comment (deluan) [inline ui/src/common/RatingField.js:10]: Can you add `color: 'inherit'`, to make it "blend" better with the theme?

File: ui/src/common/RatingField.js [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+  resource: PropTypes.string.isRequired,
+  record: PropTypes.object.isRequired,
+  visible: PropTypes.bool,
+  size: PropTypes.string,
+}
+
+RatingField.defaultProps = {
+  record: {},
+  visible: true,
+  size: 'medium',
```
Comment (deluan) [inline ui/src/common/RatingField.js:61]: I think the default should be "small", to be less "aggressive" when used as a column in the DataGrids

**Review by deluan — CHANGES_REQUESTED** [review]
This looks good! Seems to be working fine. Besides the couple of small issues in the comments below, you should also add translations for the `rating` columns in the `en.json` file

**Comment by deluan** [comment]
Can you also add a "Top Rated" Sub Menu in the Albums List? Take a look at the `ui/src/album/albumList.js` file. Add it after the "starred" option. It will also require a translation here: https://github.com/navidrome/navidrome/blob/ac37bf3631610747001f44fe843517f3fdf0ac08/ui/src/i18n/en.json#L63

File: ui/src/album/albumLists.js [hunk 3]
```diff
@@ -18,6 +19,12 @@ export default {
       params: 'sort=starred_at&order=DESC&filter={"starred":true}',
     },
   }),
+  ...(config.enableStarRating && {
+    rating: {
+      icon: StarIcon,
+      params: 'sort=rating&order=DESC',
```
Comment (deluan) [inline ui/src/album/albumLists.js:25]: I just added a `has_rating` filter in the album_repository, you can set it with:
```
&filter={"has_rating":true)
```
So only rated albums will show up in this view

File: ui/src/album/albumLists.js [hunk 4]
```diff
@@ -18,6 +19,12 @@ export default {
       params: 'sort=starred_at&order=DESC&filter={"starred":true}',
     },
   }),
+  ...(config.enableStarRating && {
+    rating: {
```
Comment (deluan) [inline ui/src/album/albumLists.js:23]: This should be called `topRated`

**Comment by neilchauhan2** [comment]
@deluan I've made all the improvements mentioned by you in the above comments, as well as included the empty stars icon. Please review the PR.

**Comment by deluan** [comment]
Hey @neilchauhan2 , looks really good, thanks! One small thing slipped through the cracks: In the `AlbumSongs.js`, the `RatingField` should have `sortable={false}`, or else it make the column sortable, something we don't want in this view

**Comment by deluan** [comment]
You should also change the `sortByOrder` attribute to `DESC` in `SongList.js` (and probably in all other lists), so then when a user sorts the list by rating, the top rated appears first

File: ui/src/i18n/en.json [hunk 5]
```diff
@@ -20,6 +20,7 @@
         "bitRate": "Bit rate",
         "discSubtitle": "Disc Subtitle",
         "starred": "Favourite",
+        "rating": "Rating",
```
Comment (deluan) [inline ui/src/i18n/en.json:23]: I think you are still missing the `rating` field for albums and artists in this file.

**Review by deluan — CHANGES_REQUESTED** [review]

File: ui/src/album/AlbumListView.js [hunk 6]
```diff
@@ -120,6 +120,7 @@ const AlbumListView = ({
               <br />
               <RatingField
                 record={r}
+                sortByOrder={'DESC'}
```
Comment (deluan) [inline ui/src/album/AlbumListView.js:123]: Here it is not necessary, it is only used for making the header clickable in the DataGrid, not in SimpleLists

**Review by deluan — CHANGES_REQUESTED** [review]

**Comment by neilchauhan2** [comment]
@deluan I'll make these changes in some time, till then you can check if there is anything else that needs to be improved.

**Comment by deluan** [comment]
Ok, will take a throughly testing and will let you know, thanks a lot for this!

**Comment by deluan** [comment]
> @deluan I'll make these changes in some time, till then you can check if there is anything else that needs to be improved.

I think that's it. After removing `sortByOrder={'DESC'}` from SimpleLists, we are good to merge!

**Comment by deluan** [comment]
I think that's it! Thanks for your effort! Merging it now

**Pull request merged**